CMS-Enterprise/sbom-harbor
# Harbor snyk crawler

## Summary
Snyk crawler to generate SBOMS from Snyk API

### Added
Merged main branch, includes migration to hyper api for https requests
Uses Snyk api to:

- Find all valid orgs in CMS
- Obtain all projects for each org
- Filter out invalid projects
- Retrieves SBOM for each project

## How to test

Run the test_get_snyk_data() unit test. Inspect logs and verify SBOMS have been added to test folder

## Coming Soon
- Send SBOMs to harbor instead of writing to file
- Better documentation
- More unit tests
- Mocks to avoid pinging Snyk api each time a unit test is run
